app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -42,6 +47,11 @@ class AccountManager @Inject constructor(db: AppDatabase) {
         private set
     private val accountDao: AccountDao = db.accountDao()
 
+    // For tracking when a media upload fails in the service
+    private val draftDao: DraftDao = db.draftDao()
+    private var draftsNeedUserAlertCurrent: LiveData<Int>? = null
+    public val draftsNeedUserAlert = MediatorLiveData<Int>()
```

Comment:
Using the mediator means that we can switch out the LiveData when we switch out users. In the current version of the feature, if you send a post then switch users very quickly, and the post fails to upload after you switched users, you will not be notified until the next time you switch back. That's sub-optimal but seemed better, in this version of the feature, than trying to negotiate switching users from the "post failed" alert pane (?) or deciding what to do in a nightmare scenario where multiple users have failed posts at the same time.

It's not clear to me AccountManager is the best place to manage the dao object here, BUT it's good in that it means we can get immediate switching to the correct user when the user changes.